api/server.js

```diff
@@ -0,0 +1,217 @@
+const express = require("express");
+const morgan = require("morgan");
+const helmet = require("helmet");
+const cors = require("cors");
+const userDb = require("../data/helpers/userDb");
+const postsDb = require("../data/helpers/postDb");
+const tagDb = require("../data/helpers/tagDb");
+
+const server = express();
+
+// MIDDLEWARE
+
+// Applying Middleware
+server.use(morgan("short"));
+server.use(helmet());
+server.use(cors());
+server.use(express.json());
+
+// Custom Middleware
+function uppercaseUsername(req, res, next) {
+  let { name } = req.body;
+
+  if (!name) {
+    return res
+      .status(400)
+      .json({ message: "Please provide a UserName, thank you" });
+  }
+  if (name.length > 128) {
+    return res.status(400).json({
+      message: "Username can not exceed 128 characters, i mean, come on!!!"
+    });
+  }
+  if (name[0] !== name[0].toUpperCase()) {
+    res.status(400).json({
+      message: "Please capitalize the first letter of the UserName, thank you."
+    });
+  } else {
+    next();
+  }
+}
+
+// ROUTES
+
+// USER ROUTES
+
+// Single User Posts List
+server.get("/api/users/:userId/posts", (req, res) => {
```

Comment:
You're one of the few people I've seen make their url nested like this. I think this is what you should do since the posts are tied to the users anyway, good job.